webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/MainCampaignService.scala

```diff
@@ -174,14 +174,15 @@ class MainCampaignService(repo: CampaignEventRepository, campaignRepo: CampaignR
                         event.copy(state = Skipped(s"Event was cancelled because campaign is archived. ${reasonMessage}"))
                       )
                     case Enabled                =>
-                      if (event.start.isAfter(now)) {
+                      val effectiveStart = event.start.minusHours(1)
```

Comment:
can you make these two time parameter of the `MainCampaignService` class so that it remains testable / configurable ?